lib/utils/packaging/unarchive_unix.go

```diff
@@ -0,0 +1,169 @@
+//go:build !windows
+
+/*
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package packaging
+
+import (
+	"archive/tar"
+	"compress/gzip"
+	"errors"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"slices"
+	"strings"
+
+	"github.com/google/renameio/v2"
+	"github.com/gravitational/trace"
+)
+
+// ReplaceToolsBinaries extracts the package into `extractDir` and replaces the specified
+// applications with symlinks in the tool's directory.
+func ReplaceToolsBinaries(toolsDir string, archivePath string, extractDir string, apps []string) error {
```

Comment:
```suggestion
// ReplaceToolsBinaries extracts executables specified by execNames from archivePath into extractDir.
// After each executable is extracted, it is symlinked from toolsDir/[name] to extractDir/[name].
// All execNames must be in the "teleport/" directory within the archive.
//
// For Darwin, archivePath must be must be a .pkg file.
// For other POSIX, archivePath must be a gzipped tarball.
func ReplaceToolsBinaries(toolsDir string, archivePath string, extractDir string, execNames []string) error {
```
Focus on thorough documentation for the exported functions, so that someone using this shared function understands how it works. This is more important than documenting unexported functions like replaceTarGz.

(The word "apps" is confusing to me. I thought this was referring to `.app` directories on macOS for a minute)